### DISCUSSION TITLE
Trivy terraform config scan duplicating results when using terraform modules

### DISCUSSION THREAD

**Discussion by chejn** [aquasecurity/trivy#4375] [Bugs]
## Description
Results seem to be duplicated when terraform files import modules from another file. For example, if there are three terraform files which import a module from another file containing a vulnerable configuration, 3 times the results will be generated for each finding and all which reference the same file where the module was defined.

Poc files
https://github.com/chejn/trivy-bug-example
## What did you expect to happen?
The configuration scanner to only identify the vulnerable module in the file which it is defined, and ignore files which import the module but do not define a configuration.

## What happened instead?
An alert was generated for each file which referenced the imported module thus duplicating the results.

Example of duplication
<img width="784" alt="Screen Shot 2022-09-30 at 10 27 09 am" src="https://user-images.githubusercontent.com/41510168/193164711-8662e6ba-0d9a-4475-8c49-2a282ca70ece.png">

## Output of run with `-debug`:

``` bash
@chejn ➜ /workspaces/Codespace (main ✗) $ ls ./Bug1/*
./Bug1/test1.tf  ./Bug1/test2.tf  ./Bug1/test3.tf

./Bug1/modules:
main.tf
@chejn ➜ /workspaces/Codespace (main ✗) $ cat ./Bug1/*
cat: ./Bug1/modules: Is a directory

module "asd" {
  source   = "./modules"
}

module "core" {
  source   = "./modules"
}

module "asd" {
  source   = "./modules"
}
@chejn ➜ /workspaces/Codespace (main ✗) $ cat ./Bug1/modules/main.tf 
resource "aws_s3_bucket" "test" {
  force_destroy = true
}

@chejn ➜ /workspaces/Codespace (main ✗) $ trivy config ./Bug1 --severity HIGH,CRITICAL | grep Tests
2022-09-30T00:10:31.242Z        INFO    Misconfiguration scanning is enabled
2022-09-30T00:10:31.425Z        INFO    Detected config files: 1
Tests: 19 (SUCCESSES: 1, FAILURES: 18, EXCEPTIONS: 0)
@chejn ➜ /workspaces/Codespace (main ✗) $ trivy config ./Bug1/modules/ --severity HIGH,CRITICAL | grep Tests
2022-09-30T00:10:52.927Z        INFO    Misconfiguration scanning is enabled
2022-09-30T00:10:53.105Z        INFO    Detected config files: 1
Tests: 7 (SUCCESSES: 1, FAILURES: 6, EXCEPTIONS: 0)
```

## Output of `trivy -v`:

```
@chejn ➜ /workspaces/Codespace (main ✗) $ trivy -v
Version: 0.31.2
```

**Comment by sblask** [comment]
Are you sure it's because they are imported by other modules? Because I am getting duplicates as well when running trivy on my modules which do not import one another with version 0.35.

**Comment by simar7** [comment]
Thanks for the discussion. I think having the individual cause of where the issue originated would help.

```
 Individual Causes
  - modules/main.tf:2-4 (module.asd)
  - modules/main.tf:2-4 (module.core)
  - modules/main.tf:2-4 (module.cores)
```

I've opened https://github.com/aquasecurity/trivy/issues/4581 to track this.
